src/commissaire_service/storage/__init__.py

```diff
@@ -134,18 +134,28 @@ def on_save(self, message, model_type_name, model_json_data):
         default values as needed, and saves it to a store; then returns the
         full saved JSON data.
 
+        The JSON data argument may also be a list of models of the same type,
+        which returns a list of full models; equivalent to calling the method
+        once for each list item, with fewer bus messages.
+
         :param message: A message instance
         :type message: kombu.message.Message
         :param model_type_name: Model type for the JSON data
         :type model_type_name: str
-        :param model_json_data: JSON representation of a model
-        :type model_json_data: dict or str
-        :returns: full dict representation of the model
-        :rtype: dict
+        :param model_json_data: JSON representation of one or more models
+        :type model_json_data: dict or str, or list of dict or str
+        :returns: full dict representation of the model(s)
+        :rtype: dict, or list of dicts
         """
-        model_instance = self._build_model(model_type_name, model_json_data)
-        saved_model_instance = self._manager.save(model_instance)
-        return saved_model_instance.to_dict()
+        if isinstance(model_json_data, list):
```

Comment:
Eh, I'm not a big fan of that for only 3 cases (one of which, `on_delete`, is an oddball since it doesn't return anything).  If there were, say, 5 or more methods with that exact structure I'd do it.